release-engineer-tools/prepare-signed-flasher-bundle.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/bash
+#
+# ------------------------------------------------------------------------------
+# This script is meant to automate and guard rail the work of a release engineer
+# to take an unsigned-flasher-bundle produced by the "prod.secure" build plan
+# and make it into signed-flasher-bundle that is then used by:
+# - production systems in the factory
+# - field customers to perform OTA Software Update
+#
+# The high level steps are:
+#
+# 1. Extract and validate the input
+# 2. Sign binaries used for production systems
+# 3. Generate encrypted ota for field customer upgrade
+# 4. Re-archive the updated package
+# ------------------------------------------------------------------------------
+
+# BEFORE YOU BEGIN
+# ------------------------------------------------------------------------------
+# 1. Make sure you have the HSM Key plugged in
+# 2. Make sure you remember your HSM PIN
+# 3. To re-test the connection to the HSM, you can do:
+#    gpgconf --kill all && sudo systemctl restart pcscd && pkcs11-tool -O
+
+# ------------------------------------------------------------------------------
+# Set CLI arguments to know values
+# ------------------------------------------------------------------------------
+unsetVal="unset"
+productVariantArg=$unsetVal
+buildVariantArg=$unsetVal
+versionArg=$unsetVal
+unsignedFlasherBundleArg=$unsetVal
+signedFlasherBundleArg=$unsetVal
+verboseArg=$unsetVal
+
+# Parse for CLI arguments
+# ------------------------------------------------------------------------------
+
+if [ "$#" -ne 5 ]; then
+    printf "Arguments Count: $# is insufficient\n\n"
```

Comment:
Given you're doing a `!=` check, it may make more sense to indicate that the script expects `exactly` 5 arguments, vs `at least` 5 arguments.

Reply:
Having said that, you've got the hard coded `verboseArg` below, which I'm guessing you intend to convert to an argument eventually. If that's the case, this should probably change somehow. In addition, I'm not sure how well positional arguments will work if you try to add a `-v` style option to configure verbosity as a user might do

```
tme.sifton prod.secure 0.9.4 bundle-unsigned.tar bundle-signed.tar -v
```

or

```
-v tme.sifton prod.secure 0.9.4 bundle-unsigned.tar bundle-signed.tar
```

Reply:
Fair point, I pushed a change that reworded this and ask for exactly the number of args.
I was going back on forth on weather this should be done using the `--key=value` style parser (which can be re-ordered) and decided against it since it makes it less things to type for the `release engineer`. I can be convinced otherwise, though.